src/power.jl

```diff
@@ -184,31 +211,33 @@ c_k & = & 2 \\sum_{j=0}^{(k-2)/2} a_{k-j} a_j + (a_{k/2})^2,
 """ sqr!
 
 for T = (:Taylor1, :TaylorN)
-    @eval function sqr!(c::$T, a::$T, k::Int)
-        if k == 0
-            @inbounds c[1] = a[1]^2
-            return nothing
-        end
+    @eval begin
+        @inline function sqr!(c::$T, a::$T, k::Int)
+            if k == 0
+                @inbounds c[1] = constant_term(a)^2
+                return nothing
+            end
+
+            kodd = k%2
+            kend = div(k - 2 + kodd, 2)
+            @inbounds for i = 0:kend
+                if $T == Taylor1
```

Comment:
It could *almost* be replaced by `mul!(c,a,a,kend)`, but it can't. The reason is that the core of the loop is identical to the loop in `mul!`, but it updates `c[k+1]` instead of `c[kend+1]` as `mul!` would do.